compiler/rustc_middle/src/ty/context.rs

```diff
@@ -1285,6 +1285,33 @@ impl<'tcx> TyCtxt<'tcx> {
         )
     }
 
+    /// Find the nearest parent module of a [`DefId`].
+    /// This is different from [`TyCtxt::parent()`] because this returns
+    /// the nearest parent `mod`, whereas [`TyCtxt::parent()`] returns
+    /// the nearest module-like item (e.g., the enum that a variant belongs to).
+    ///
+    /// Returns `def_id` if it's the crate root.
+    ///
+    /// Note for users in rustdoc: **panics if the item it belongs to is fake**
```

Comment:
This is kind of assumed for all compiler APIs, I don't think you need to call it out explicitly. Only rustdoc APIs *can* handle it because they're the only ones that know about fake IDs.

Reply:
Well you told me to mention that the rustdoc function that was moved here can panic in #80368 :)

Reply:
Right, because it was a rustdoc API then, and could have had either behavior. Now there's only one possibly behavior (note that Item::is_fake and all other ways of telling apart fake IDs are only in the rustdoc crate).

Reply:
Ah, I see what you mean: in general you have to assume that non-rustdoc API may panic on fake IDs. I'll remove it then.